tests/log-messages.feature

```diff
@@ -0,0 +1,62 @@
+# This file is part of REANA.
+# Copyright (C) 2023 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+Feature: Log messages
+
+    As a researcher,
+    I want to be able to see the log messages of my workflow execution,
+    So that I can verify that the workflow ran correctly.
+
+    Scenario: The workflow start has produced the expected messages
+        When the workflow is finished
+        Then the engine logs should contain "snakemake.logging | MainThread | WARNING | Building DAG of jobs..."
+
+    Scenario: The SCRAM step has produced the expected messages
+        When the workflow is finished
+        Then the engine logs should contain "reana-workflow-engine-snakemake | MainThread | INFO | Job 'scram' received"
+        And the job logs for the "scram" step should contain "Registered EDM Plugin: HiggsExample20112012HiggsDemoAnalyzer"
```

Comment:
Some of these lines are really long. What about always using triple quoting and putting those expected log values on new lines?  Typically, researchers might have long outputs, so it could be perhaps more ready for a copy/paste recipe in this way:

```
		And the job logs for the "scram" step should contain 
			"""
			Registered EDM Plugin: HiggsExample20112012HiggsDemoAnalyzer
			"""
```